src/main/java/com/adobe/epubcheck/overlay/OverlayHandler.java

```diff
@@ -135,5 +140,20 @@ public void ignorableWhitespace(char[] chars, int arg1, int arg2)
   public void processingInstruction(String arg0, String arg1)
   {
   }
-
+  
+  private void checkFragment(String ref) {
+  
+    if (ref.indexOf("#") == -1) {
+      // must include a fragid
+      report.message(MessageId.MED_014, EPUBLocation.create(path, parser.getLineNumber(), parser.getColumnNumber()));
+    }
+    
+    else {
+      String frag = PathUtil.getFragment(ref.trim());
+      if (Strings.isNullOrEmpty(frag)) {
+        // empty fragid # not allowed
+        report.message(MessageId.MED_015, EPUBLocation.create(path, parser.getLineNumber(), parser.getColumnNumber()));
+      }
+    }
```

Comment:
If we want to only create one message for this (see the comment below), we can combine the logic with something like:

```java
    if (ref.indexOf("#") == -1
         || Strings.isNullOrEmpty(PathUtil.getFragment(ref.trim()))) {
        // must include a non-empty fragment identifier
        report.message(MessageId.MED_014, EPUBLocation.create(path, parser.getLineNumber(), parser.getColumnNumber()));
      }
```